### ISSUE TITLE
Add build support of S2I build

### ISSUE THREAD

**Issue by surajssd** [kedgeproject/kedge#309]
This is again a specific use case that solves the build story https://github.com/kedgeproject/kedge/issues/273 for kedge using OpenShift S2I.

- When provider is OpenShift then do builds using S2I
- Generate `buildconfig` and `imagestream`

**Comment by surajssd** [comment]
Right now this is blocked on the openshift support https://github.com/kedgeproject/kedge/pull/299 and https://github.com/kedgeproject/kedge/pull/288

**Comment by surajssd** [comment]
Also before doing this, we should implement https://github.com/kedgeproject/kedge/issues/331

**Comment by kadel** [comment]
First version of this should allow to set image name and builder image

```kedge build --context directory --image output-image --builder-image builderImage:tag```

this command should generate something similar to this:

```yaml

apiVersion: v1
kind: ImageStream
metadata:
  name: output-image
spec:
  tags:
  - name: latest

---
kind: BuildConfig
apiVersion: v1
metadata:
  name: output-image
spec:
  strategy:
    type: Binary
    sourceStrategy:
      from:
        kind: DockerImage
        name: builderImage:tag
  output:
    to:
      kind: "ImageStreamTag"
      name: "output-image:latest"
```

and than run 
```oc start-build output-image --follow --from-dir directory```

**Comment by surajssd** [comment]
It is hard for a developer to know the builder image, so not sure about the flag where user provides the builder image.


> I don't think that doing git autodetection is wise. It has a lot of problems as we saw in Kopose. It would rather do builds with binary source strategy

What do you mean by git autodetection? I think what is done right now in the code is similar to what openshift/oc does.


> use oc start-build to stream local directory to openshift buildconfig.

We can even be alias to the `oc new-build ...` and `oc start-build ...`, we don't even need to create the configs. For the repo: https://github.com/surajssd/localbuild I can run following commands to get the code being built using s2i source strategy:

```
oc new-build .
oc start-build localbuild --from-dir .
```

**Comment by kadel** [comment]
closed in #452
